Added an enhancement - issue 366 to add a download all if no argument…

rabroldan opened the pull request:
This updated version automatically gets all blob fields from self.datasource.fields and adds them to the list of fields that will be downloaded if the fields argument is empty. As a result, the function will by default download all of the blob's fields if we don't specify any specific fields as parameters.

Adding an ```if condition``` to add the code that set the fields 

Kindly please rake a look thank you

kbolashev commented on dagshub/data_engine/model/query_result.py:
```diff
@@ -208,21 +208,26 @@ def __getitem__(self, item: Union[str, int, slice]):
+
+        # If no fields are specified, include all blob fields from self.datasource.fields
+        if not fields:
+            fields = [field.name for field in self.datasource.fields if field.is_blob()]
```
There is no `is_blob()` function on the `MetadataFieldSchema` enum.
You can either add this function, or compare to MetadataFieldType.BLOB in the comprehension

```suggestion
            fields = [field.name for field in self.datasource.fields if field.valueType == MetadataFieldType.BLOB]
```

Don't forget to import MetadataFieldType, I don't think it's imported in QueryResult.

kbolashev reviewed (changes requested):
Looking good! Need to change a thing though.

Also apologies for not writing this anywhere like a CONTRIBUTING.md (we should probably set up one), but we use [black](https://github.com/psf/black) to format the code.

rabroldan commented:
hi @kbolashev Kindly please review. I added and chaged the valeu as requested and added an import ```from dagshub.data_engine.dtypes import MetadataFieldType``` also reformatted the code using python black with the added codes into it.

kbolashev commented:
Thank you so much @rabroldan !
Ironing out some stuff still, can I ask you to rerun black with `-l 120`?

I'll make sure to add it to `pyproject.toml`, and also clear it up in the whole repo so it's consistent.

kbolashev commented:
Ok I reran it and resolved the merge conflict, should be able to merge it now.
Thank you for the contribution!

rabroldan commented:
hey @kbolashev i jsut saw the request. but it seemed you already did it, thank you for allowing me to contribute on this project. i'll keep an eye on some of the issue and see if i can get a few more done.
